### PR TITLE
Remove Ingress v1beta1 specific code

### PR DESCRIPTION
**Pull request by devguyio**
### Type of change

- Refactoring

### Description

Remove the code specific to v1beta1/Ingress APIs and refactor existing code to use v1/Ingress by default.

Part of #7320

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] ~Write tests~
- [x] Make sure all tests pass
- [ ] ~Update documentation~
- [ ] ~Check RBAC rights for Kubernetes / OpenShift roles~
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] ~Update CHANGELOG.md~
- [ ] ~Supply screenshots for visual changes, such as Grafana dashboards~

### CODE REVIEW

**Comment by devguyio** [comment]
This is to validate the ITs are running after removing the v1beta1 Ingress, will update it once I'm sure everything is passing and #7393 is merged.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenersReconciler.java [hunk 1]
```diff
@@ -242,9 +235,6 @@ protected Future<Void> routes() {
      * @return  Future which completes when all ingresses are created or deleted.
      */
     protected Future<Void> ingresses() {
-        if (!pfa.hasIngressV1()) {
-            return Future.succeededFuture();
-        }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenersReconciler.java:238]: Should we remove the empty line here?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenerReconcilerIngressV1Beta1Test.java [hunk 2]
```diff
@@ -227,17 +129,14 @@ public MockKafkaListenersReconciler(
...
-                IngressV1Beta1Operator ingressV1Beta1Operator) {
-            super(reconciliation, kafka, null, pfa, 300_000L, secretOperator, serviceOperator, routeOperator, ingressOperator, ingressV1Beta1Operator);
+                IngressOperator ingressOperator) {
+            super(reconciliation, kafka, null, pfa, 300_000L, secretOperator, serviceOperator, routeOperator, ingressOperator);
         }
 
         @Override
         public Future<ReconciliationResult> reconcile()  {
             return ingresses()
-                    .compose(i -> ingressesV1Beta1())
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenerReconcilerIngressV1Beta1Test.java:238]: I think we need to think about this test class. Its main purpose is that with Kube 1.16 it uses the v1beta1 methods and with Kube 1.22 it is using the v1 methods. It does not seem to check anything else. After removing the v1beta1 support, I don't think the v1 test makes any sense now. So maybe we should remove it completely?

**Review by scholzj — COMMENTED** [review]
Some nits, LGTM otherwise.

**Review by scholzj — APPROVED** [review]
Thanks. LGTM.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**